File: LeetCode/Merge_sorted_Array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int writeIter = (m+n)-1;
        int secondIter = n;
        int firstIter = m;
        secondIter--;
        firstIter--;
        while(firstIter>=0 &&secondIter>=0){
            if(nums1[firstIter] > nums2[secondIter]){
                //cout<<" writing "<<nums1[firstIter]<<" into the finalArray"<<endl;
                nums1[writeIter] = nums1[firstIter];
                firstIter--;
                writeIter--;
            }
            else{
                //cout<<" writing "<<nums2[secondIter]<<" into the finalArray"<<endl;
                nums1[writeIter] = nums2[secondIter];
                secondIter--;
                writeIter--;
            }
        }
        if(firstIter < 0){
            while(secondIter>=0){
                //cout<<" writing "<<nums2[secondIter]<<" into the finalArray"<<endl;
                nums1[secondIter] = nums2[secondIter];
                secondIter--;
            }
        }
    }
};

File: LeetCode/merge_sorted_array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int writeIter = m+n-1;
        
        int read1 = m-1;
        int read2 = n-1;
        
        while(read1 >=0 && read2 >= 0){
            if(nums1[read1] > nums2[read2]){
                nums1[writeIter] = nums1[read1];
                read1--;
            }else{
                nums1[writeIter] = nums2[read2];
                read2--;
            }
            writeIter--;
        }
        
        while(read1 >= 0){
            nums1[writeIter] = nums1[read1];
            writeIter--;
            read1--;
        }
        while(read2 >=0){
            nums1[writeIter] = nums2[read2];
            writeIter--;
            read2--;
        }
    }
}